### PR TITLE
Convert image chooser modal to generic chooser views

### PR DESCRIPTION
**Pull request by gasman**
along with various other bits of cleanup, including moving a whole lot more of the filter form logic into the base generic view.

Part of #8480; incorporates #8803

### CODE REVIEW

File: wagtail/admin/forms/choosers.py [hunk 1]
```diff
@@ -44,3 +47,90 @@ class EmailLinkChooserForm(forms.Form):
...
+
+    def filter(self, objects):
+        collection_id = self.cleaned_data.get("collection_id")
+        if collection_id:
+            self.is_filtering_by_collection = True
+            objects = objects.filter(collection=collection_id)
+        return super().filter(objects)
+
+
+class LocaleFilterMixin(forms.Form):
```
Comment (zerolab) [inline wagtail/admin/forms/choosers.py:113]: Was going to note this was outside the scope of the PR, but it is integral to letting the filter form do its thing.

File: wagtail/images/templates/wagtailimages/chooser/results.html [hunk 2]
```diff
@@ -1,28 +1,21 @@
...
                 </a>
             </li>
         {% endfor %}
     </ul>
+{% endblock %}
 
-    {% include "wagtailadmin/shared/pagination_nav.html" with items=images linkurl='wagtailimages:chooser_results' %}
-{% endif %}
+{% block no_items_message %}
+    <p>{% trans "No images have been uploaded." %}</p>
```
Comment (zerolab) [inline wagtail/images/templates/wagtailimages/chooser/results.html:20]: question: should this not be consistent with the docs `no_items_message`? https://github.com/wagtail/wagtail/blob/8ae253a47f1fc68f7a55c92481a5293227d890a2/wagtail/documents/templates/wagtaildocs/chooser/results.html#L10-L23

File: docs/releases/4.0.md [hunk 3]
```diff
@@ -162,3 +162,7 @@ If your code contains references to URL route names within the `wagtaildocs` or
 * `wagtaildocs:chooser_upload` is now `wagtaildocs_chooser:create`
 * `wagtailsnippets:list`, `wagtailsnippets:list_results`, `wagtailsnippets:add`, `wagtailsnippets:edit`, `wagtailsnippets:delete-multiple`, `wagtailsnippets:delete`, `wagtailsnippets:usage`, `wagtailsnippets:history`: These now exist in a separate `wagtailsnippets_{app_label}_{model_name}` namespace for each snippet model, and no longer take `app_label` and `model_name` as arguments.
 * `wagtailsnippets:choose`, `wagtailsnippets:choose_results`, `wagtailsnippets:chosen`: These now exist in a separate `wagtailsnippetchoosers_{app_label}_{model_name}` namespace for each snippet model, and no longer take `app_label` and `model_name` as arguments.
+
```
Comment (zerolab) [inline docs/releases/4.0.md:165]: Needs a note about the URL name changes (`wagtailimages:chooser` -> `wagtailimages_chooser:choose` etc)

**Review by zerolab — CHANGES_REQUESTED** [review]
Great use of atomic commits and step by step unpicking of the work -- made it more digestible. Still quite a few changes.

Left a suggestion on the "no results" messages and reminder for the upgrade considerations regarding the URLs changes

File: docs/releases/4.0.md (hunk 3 above)
Comment (gasman) [reply docs/releases/4.0.md]: Good catch! Added in d3ebd24741aeb4c258af785ef7688f43ae3c5af8.

File: wagtail/images/templates/wagtailimages/chooser/results.html (hunk 2 above)
Comment (gasman) [reply wagtail/images/templates/wagtailimages/chooser/results.html]: 👍  Added in 01152eb207e4e4492ea92f80feaa21f7a8dab33c.

**Review by zerolab — APPROVED** [review]
:shipit:

**Pull request merged**